server/src/main/java/com/paypal/selion/utils/SauceConfigReader.java

```diff
@@ -52,10 +53,13 @@
     private int sauceRetry = DEFAULT_RETRY_COUNT;
 
     /**
-     * @return - A {@link SauceConfigReader} object that can be used to retrieve values from the Configuration object as
-     *         represented by the JSON file
+     * @return a {@link SauceConfigReader} object that can be used to retrieve values from the Configuration object as
+     *         represented by the JSON file. Throws a {@link GridConfigurationException} on instance load error.
      */
     public static SauceConfigReader getInstance() {
+        if (reader == null) {
```

Comment:
You need to synchronize here for thread safety or document this is not thread safe.